sdk/observability/telemetry/src/main/scala/com/daml/telemetry/OpenTelemetryOwner.scala

```diff
@@ -15,76 +16,76 @@ import scala.jdk.CollectionConverters.SeqHasAsJava
 
 object OpenTelemetryOwner {
 
+  /** Add views to providers
+    *
+    * In open telemetry, you have to define the histogram views separately from the metric itself. Even worse,
+    * you need to define it before you define the metrics. If you define two views that match to the same metrics,
+    * you end up with ugly warning messages and errors: https://opentelemetry.io/docs/specs/otel/metrics/sdk/#measurement-processing
+    *
+    * The solution to this is to statically define all the metric names in advance separately, create appropriate views
+    * and then, on each histogram definition check that an appropriate static definition exists.
+    */
   def addViewsToProvider(
       builder: SdkMeterProviderBuilder,
-      histograms: Seq[HistogramDefinition],
+      configs: Seq[HistogramDefinition],
+      metrics: Set[MetricName],
   ): SdkMeterProviderBuilder = {
-    // Only one view is going to be applied, and it's in the order of it's definition
-    // therefore the config views must be registered first to be able to override the code defined views
-    val builderWithCustomViews = histograms.foldRight(builder) { case (histogram, builder) =>
-      builder.registerView(
-        histogramSelectorByName(histogram.name),
-        explicitHistogramBucketsView(histogram.bucketBoundaries),
-      )
-    }
-    builderWithCustomViews
-      // use smaller buckets for the executor services (must be declared before the generic timing buckets
-      .registerView(
-        histogramSelectorByName(
-          s"${ExecutorServiceMetrics.Prefix}*${OpenTelemetryTimer.TimerUnitAndSuffix}"
-        ),
-        explicitHistogramBucketsView(
-          Seq(
-            0.0005d, 0.001d, 0.002d, 0.005d, 0.01d, 0.025d, 0.05d, 0.1d, 0.25d, 0.5d, 0.75d, 1d,
-            2.5d,
-          )
+
+    val candidates = configs ++ Seq(
+      // use smaller buckets for the executor services (must be declared before the generic timing buckets)
+      HistogramDefinition(
+        s"${ExecutorServiceMetrics.Prefix}*${OpenTelemetryTimer.TimerUnitAndSuffix}",
+        Seq(
+          0.0005d, 0.001d, 0.002d, 0.005d, 0.01d, 0.025d, 0.05d, 0.1d, 0.25d, 0.5d, 0.75d, 1d, 2.5d,
         ),
-      )
+      ),
       // timing buckets for gRPC server latency measurements with more precise granularity on latencies up to 5s
-      .registerView(
-        histogramSelectorByName(
-          s"${DamlGrpcServerMetrics.GrpcServerMetricsPrefix}*${OpenTelemetryTimer.TimerUnitAndSuffix}"
-        ),
-        explicitHistogramBucketsView(
-          Seq(
-            0.01d, 0.025d, 0.050d, 0.075d, 0.1d, 0.15d, 0.2d, 0.25d, 0.35d, 0.5d, 0.75d, 1d, 1.25d,
-            1.5d, 1.75d, 2d, 2.25d, 2.5d, 2.75d, 3d, 3.5d, 4d, 4.5d, 5d, 10d,
-          )
+      HistogramDefinition(
+        s"${DamlGrpcServerMetrics.GrpcServerMetricsPrefix}*${OpenTelemetryTimer.TimerUnitAndSuffix}",
```

Comment:
Just double checking here, is there a way we can override this one? 
If not can we change the logic to register in such a way that it first goes through the config and if it doesn't find anything there then it goes through these provided candidates?

Reasoning:
We recently had a scenario where we wanted more detailed gRPC buckets as our latency was much higher than the max 10s and we wanted to see improvements there.